access/handler.go

```diff
@@ -701,6 +713,192 @@ func (h *Handler) GetExecutionResultByID(ctx context.Context, req *access.GetExe
+// block as it becomes available.
+//
+// Each lightweight block are filtered by the provided block status, and only
+// those blocks that match the status are returned.
+func (h *Handler) SubscribeBlockDigests(request *access.SubscribeBlockDigestsRequest, stream access.AccessAPI_SubscribeBlockDigestsServer) error {
+	// check if the maximum number of streams is reached
+	if h.StreamCount.Load() >= h.MaxStreams {
+		return status.Errorf(codes.ResourceExhausted, "maximum number of streams reached")
+	}
+	h.StreamCount.Add(1)
+	defer h.StreamCount.Add(-1)
+
+	startBlockID, startBlockHeight, blockStatus, err := h.getStartData(request.GetStartBlock(), request.GetBlockStatus())
+	if err != nil {
+		return status.Errorf(codes.InvalidArgument, "invalid argument: %v", err)
+	}
+
+	sub := h.api.SubscribeBlockDigests(stream.Context(), startBlockID, startBlockHeight, blockStatus)
+	for {
+		v, ok := <-sub.Channel()
+		if !ok {
+			if sub.Err() != nil {
+				return rpc.ConvertError(sub.Err(), "stream encountered an error", codes.Internal)
+			}
+			return nil
+		}
+
+		blockDigest, ok := v.(*flow.BlockDigest)
+		if !ok {
+			return status.Errorf(codes.Internal, "unexpected response type: %T", v)
+		}
+
+		err = stream.Send(&access.SubscribeBlockDigestsResponse{
+			BlockId:        convert.IdentifierToMessage(blockDigest.ID),
+			BlockHeight:    blockDigest.Height,
+			BlockTimestamp: timestamppb.New(blockDigest.Timestamp),
+		})
+		if err != nil {
+			return rpc.ConvertError(err, "could not send response", codes.Internal)
+		}
+	}
+}
+
+// getStartData processes subscription start data.
+// It takes a byte slice representing the start block ID and a BlockStatus from the entities package.
+// The method returns a flow.Identifier representing the start block ID, a flow.BlockStatus representing the block status,
+// and an error if any issues are encountered during the processing.
+//
+// Returns:
+// - flow.Identifier: The start block id for searching.
+// - uint64: The start block height for searching.
+// - flow.BlockStatus: Block status.
+// - error: An error indicating the result of the operation, if any.
+// Errors:
+// - codes.InvalidArgument: If blockStatus is flow.BlockStatusUnknown, or startBlockID could not convert to flow.Identifier.
+func (h *Handler) getStartData(msg *access.StartBlock, msgBlockStatus entities.BlockStatus) (flow.Identifier, uint64, flow.BlockStatus, error) {
+	var err error
+	var startBlockID = flow.ZeroID
+	var startBlockHeight uint64 = 0
+
+	switch s := msg.StartBlock.(type) {
+	case *access.StartBlock_BlockId:
+		startBlockID, err = convert.BlockID(s.BlockId)
+		if err != nil {
+			return flow.ZeroID, 0, flow.BlockStatusUnknown, fmt.Errorf("invalid start block ID argument: %w", err)
+		}
+	case *access.StartBlock_BlockHeight:
+		startBlockHeight = s.BlockHeight
+	}
+
+	blockStatus := convert.MessageToBlockStatus(msgBlockStatus)
+	err = checkBlockStatus(blockStatus)
```

Comment:
This must return `codes.InvalidArgument` to comply with the godoc, but it doesn't